lib/web_of_science/harvester.rb

```diff
@@ -0,0 +1,145 @@
+module WebOfScience
+
+  # Application logic to harvest publications from Web of Science;
+  # This is the bridge between the WebOfScience API and the SUL-PUB application.
+  # This class is responsible for processing WebOfScience API response data
+  # to integrate it into the application data models.
+  class Harvester
+
+    # Harvest all publications for an author
+    # @param author [Author]
+    # @return [Array<String>] WosUIDs that create Publications
+    def process_author(author)
+      raise(ArgumentError, 'author must be an Author') unless author.is_a? Author
+      process_records author, records_for_author(author)
+    end
+
+    # Harvest DOI publications for an author
+    # @param author [Author]
+    # @param dois [Array<String>] DOI values (not URIs)
+    # @return [Array<String>] WosUIDs that create Publications
+    def process_dois(author, dois)
+      raise(ArgumentError, 'author must be an Author') unless author.is_a? Author
+      raise(ArgumentError, 'dois must be Enumerable') unless dois.is_a? Enumerable
+      # TODO: normalize the dois using altmetrics identifier gem, as in PR #246
+      dois.reject! { |doi| publication_identifier?('doi', doi) }
+      return [] if dois.empty?
+      records = wos_queries.search_by_doi(dois.shift)
+      dois.each { |doi| records.merge_records wos_queries.search_by_doi(doi) }
+      process_records author, records
+    end
+
+    # Harvest PMID publications for an author
+    # @param author [Author]
+    # @param pmids [Array<String>] PMID values (not URIs)
+    # @return [Array<String>] WosUIDs that create Publications
+    def process_pmids(author, pmids)
+      raise(ArgumentError, 'author must be an Author') unless author.is_a? Author
+      raise(ArgumentError, 'pmids must be Enumerable') unless pmids.is_a? Enumerable
+      # TODO: normalize the pmids using altmetrics identifier gem, as in PR #246
+      pmids.reject! { |pmid| publication_identifier?('pmid', pmid) }
+      return [] if pmids.empty?
+      # TODO: define the flow for retrieving this data
+      # PMID ->  links-API to get UID -> WOS API to get record (likely MEDLINE record)
+      #      -> (if that fails) PubMed API for PubMed record
+      # TODO: process_records author, wos_queries.retrieve_by_id(pmids) # Does this work?
+      author.id # this is only here to avoid a rubocop error
```

Comment:
so this method is not actually intended to be implemented and returns nothing?  is that why there is a "raise" at the end?  and curious as to what the rubocop error is?

Reply:
Let's ticket the work on PMID harvesting via the new WOS-API when we are starting out with a PMID and we do not have a WOS record yet.  The process flow for this needs to be worked out.  Nevertheless, this is the placeholder for this, to declare an intention to support this public API.

Reply:
https://github.com/sul-dlss/sul_pub/issues/408